tools/debug_tools/crash_log_parser/crash_log_parser.py

```diff
@@ -0,0 +1,266 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+from os import path
+import re, bisect
+from subprocess import check_output
+import sys
+
+#arm-none-eabi-nm -nl <elf file>
+NM_EXEC = "arm-none-eabi-nm"
+OPT = "-nlC"
+ptn = re.compile("([0-9a-f]*) ([Tt]) ([^\t\n]*)(?:\t(.*):([0-9]*))?")
+fnt = None
+fnt = ""
+
+class ElfHelper(object):
+    def __init__(self, p,m):
+        if path.isfile(p):
+            elf_path = p
+        if path.isfile(m):
+            map_path = m
+        
+        op = check_output([NM_EXEC, OPT, elf_path])
+        map_file = open(map_path)
+        self.maplines = map_file.readlines()
+        self.matches = ptn.findall(op)
+        self.addrs = [int(x[0],16) for x in self.matches]
+        #print(self.maplines)
+        
+    def function_addrs(self):
+        return self.addrs
+    
+    def function_name_for_addr(self, addr):
+        i = bisect.bisect_right(self.addrs, addr)
+        funcname = self.matches[i-1][2]
+        return funcname
+
+    def file_name_for_function_name(self, funcname):
+        for eachline in self.maplines:
+            #print("%s:%s"%(eachline,funcname))
```

Comment:
dead code should be removed in this file

Reply:
Thanks for the review, and will fix the comments. After some discussion here and getting some feed backs, I'm going to make this a build time enabled feature. We will also make this depend on if TARGET_CORTEX_M is defined so that it doesn't fail on Cortex-A targets.